BlueprintUI/Sources/Layout/LayoutSubelement.swift

```diff
@@ -1,8 +1,189 @@
 import CoreGraphics
+import Foundation
+import QuartzCore
 
-
+/// A collection of proxy values that represent the child elements of a layout.
 public typealias LayoutSubelements = [LayoutSubelement]
 
+/// A proxy that represents one child element of a layout.
+///
+/// This type acts as a proxy for a child element in a ``Layout``. Layout protocol methods receive a
+/// ``LayoutSubelements`` collection that contains exactly one proxy for each of the child elements
+/// managed by the layout.
+///
+/// Use this proxy to get information about the associated element, like its size and traits. You
+/// should also use the proxy to tell its corresponding element where to appear by calling the
+/// proxy’s ``place(at:anchor:size:)`` method. Do this once for each subview from your
+/// implementation of the layout’s ``CaffeinatedLayout/placeSubelements(in:subelements:cache:)``
+/// method.
+///
 public struct LayoutSubelement {
-    // TODO:
+
+    typealias SizeCache = HintingSizeCache
+
+    var identifier: ElementIdentifier
+    private var content: ElementContent
+    var measureContext: MeasureContext
+
+    var environment: Environment { measureContext.environment }
+    private var cache: HintingSizeCache { measureContext.node.sizeCache }
+
+    @Storage
```

Comment:
Probably for later; but I'd be curious on if there's any performance benefit to the property wrappers (which are each their own instance), vs. converting the `LayoutSubelement` itself into a `final class`.